File: Kasa/NSColor+CGColor.m
//
//  NSColor+CGColor.m
//  Kasa
//
//  Created by Luavis on 12. 11. 5..
//  Copyright (c) 2012년 Luavis. All rights reserved.
//

#import "NSColor+CGColor.h"

@implementation NSColor (CGColor)

- (CGColorRef)CGColor
{
    const NSInteger numberOfComponents = [self numberOfComponents];
    CGFloat components[numberOfComponents];
    CGColorSpaceRef colorSpace = [[self colorSpace] CGColorSpace];
    
    [self getComponents:(CGFloat *)&components];
    
    return CGColorCreate(colorSpace, components);
}

+ (NSColor *)colorWithCGColor:(CGColorRef)CGColor
{
    if (CGColor == NULL) return nil;
    return [NSColor colorWithCIColor:[CIColor colorWithCGColor:CGColor]];
}

@end

File: Kasa.swift/NSColor+CGColor.swift
//
//  NSColor+CGColor.swift
//  Kasa.swift
//
//  Created by Luavis on 2014. 9. 8..
//  Copyright (c) 2014년 Luavis. All rights reserved.
//

import Foundation

extension NSColor {
    func CGColor() -> CGColorRef? {
        let numberOfComponents: NSInteger = self.numberOfComponents
        let components: UnsafeMutablePointer<CGFloat> = UnsafeMutablePointer<CGFloat>(calloc(UInt(numberOfComponents), UInt(sizeof(CGFloat))))
        var colorSpace:CGColorSpaceRef = self.colorSpace.CGColorSpace;
        
        self.getComponents(components)
        return CGColorCreate(colorSpace, components)
    }
    
    class func colorWithWithCGColor(CGColor: CGColorRef!) -> NSColor {
        return NSColor.colorWithWithCGColor(CGColor)
    }
}
